classes/local/groups.php

```diff
@@ -68,16 +68,34 @@ public function __construct($apirest = null, $rand = null) {
      * @return array[stdClass] $groups
      */
     public function get_groups() {
-        $response = $this->apirest->get_groups(10000, 1);
-        if (!isset($response->groups)) {
-            throw new \moodle_exception('getgroupserror', 'accredible', 'https://help.accredible.com/hc/en-us');
-        }
+        $pagesize = 50;
+        $page = 1;
+
+        // Maximum number of pages to request to avoid possible infinite loop.
+        $looplimit = 100;
+        try {
+            $loop = true;
+            $count = 0;
+            $groups = array();
+            // Query the Accredible API and loop until it returns that there is no next page.
+            while ($loop === true) {
```

Comment:
It might be better to use `for ($i = 0; $i <= 100; $i++)` here as we don't have to check for `$count >= $looplimit` below.